File: asakatsu/20221022/3.rs
// https://atcoder.jp/contests/abc001/tasks/abc001_2

// attributes
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(non_snake_case)]

// imports
use std::collections::{HashMap, BTreeMap, VecDeque, BinaryHeap};
use std::cmp::Reverse;

// input macro
// [Rustで競技プログラミング スターターキット](https://qiita.com/hatoo@github/items/fa14ad36a1b568d14f3e)
macro_rules! get {
    ($t:ty) => {
        {
            let mut line = String::new();
            std::io::stdin().read_line(&mut line).unwrap();
            line.trim().parse::<$t>().unwrap()
        }
    };
    ($($t:ty),*) => {
        {
            let mut line = String::new();
            std::io::stdin().read_line(&mut line).unwrap();
            let mut iter = line.split_whitespace();
            (
                $(iter.next().unwrap().parse::<$t>().unwrap(),)*
            )
        }
    };
    ($t:ty ; $n:expr) => {
        (0..$n).map(|_|
            get!($t)
        ).collect::<Vec<_>>()
    };
    ($($t:ty),* ; $n:expr) => {
        (0..$n).map(|_|
            get!($($t),*)
        ).collect::<Vec<_>>()
    };
    ($t:ty ;;) => {
        {
            let mut line = String::new();
            std::io::stdin().read_line(&mut line).unwrap();
            line.split_whitespace()
                .map(|t| t.parse::<$t>().unwrap())
                .collect::<Vec<_>>()
        }
    };
    ($t:ty ;; $n:expr) => {
        (0..$n).map(|_|
            get!($t ;;)
        ).collect::<Vec<_>>()
    };
}


// solve
fn main() {
    let m = get!(usize);

    if m < 100 {
        println!("00");
    } else if 100 <= m && m <= 5_000 {
        println!("{:0>2}", m / 100);
    } else if 6_000 <= m && m < 30_000 {
        println!("{}", m / 1_000 + 50);
    } else if 3_500 <= m && m <= 70_000 {
        println!("{}", (m/1_000 - 30)/5 + 80);
    } else if m > 70_000 {
        println!("89");
    }
}

File: virtual_contest/Mayo20230302/3.py
# https://atcoder.jp/contests/abc136/tasks/abc136_d

# よくわかんないからダブリング

from collections import Counter

LOG = 333  # 2^333 > 10^100

S = input()
N = len(S)

# ダブリング配列
# dp[i][j] := i文字目の人が2^j回移動したときにいる場所
dp = [[0] * N for _ in range(LOG)]

for j, d in enumerate(S):
    if d == "R":
        dp[0][j] = j + 1
    else:
        dp[0][j] = j - 1

for i in range(1, LOG):
    for j in range(N):
        dp[i][j] = dp[i-1][dp[i-1][j]]

# カウント
cnt = Counter(dp[-1])

for i in range(N):
    print(cnt[i], end=" ")
print()
